metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py

```diff
@@ -541,6 +544,48 @@ def on_dag_start(self, dag_run: "DagRun") -> None:
 
             self.emitter.emit(event)
 
+        if dag.dag_id == _DATAHUB_CLEANUP_DAG:
+            assert self.graph
+
+            logger.debug("Initiating the cleanup of obsselete data from datahub")
+
+            ingested_dataflow_urns = list(
+                self.graph.get_urns_by_filter(
```

Comment:
I think you should filter for `cluster` as well; otherwise if user has multiple Airflow instance you will delete dags which you shouldn't.

Reply:
I am filtering the entire URN which is already having the `cluster` i.e. `urn:li:dataFlow:(airflow,simple_dag,prod)`
So, still we need to match/filter `cluster` explicitly? or my understanding is wrong.

Reply:
If you check here we use cluster or env to generate the DataFlow Urns, so it is part of the urn. ->
https://github.com/datahub-project/datahub/blob/90febde88e6993bebe38e368597f90855b966bd7/metadata-ingestion/src/datahub/api/entities/datajob/dataflow.py#L75

This means if the env or cluster is set and has multiple Airflow environments like `DEV` and `PROD`, then your query will return the urns for both PROD and DEV, which we don't want in this case as these are different Airflow environment.

You should add cluster/env as a filter parameter.